### PR TITLE
[VWIP] Create migrating_from_eeglab.rst

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Start docs for migrating from EEGLAB.

(I've never used .rst before.)

@jasmainak

### CODE REVIEW

**Comment by jasmainak** [comment]
Awesome, did you try building? 

`make html-dev_noplots`

I will pull and play with it.

File: doc/manual/migrating_from_eeglab.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Caveats coming from EEGLAB
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: you need to create a link to this file in `doc/manual/index.rst` :)

File: doc/manual/migrating_from_eeglab.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* does not have the extensive inbuilt *single-trial* capabilities of EEGLAB. For example, while it is possible to plot single trials sorted by alpha power by feeding mne.viz.plot_epochs_image a function built around psd_epochs, it requires much more user input.
+* does not easily allow the rich storage of epochs-level *event information* EEGLAB has. MNE events are stored in the three-column `events` format, all integers.
+* does not have an equivalent to EEGLAB's *STUDY* structure. Grand averages of single-subject `evoked` and `average_tfr` objects can be constructed with the `mne.grand_average` function (beware of the default setting of weighting by trials), which return regular `evoked` and `average_tfr` objects. Individual subjects can simply be stored in lists or dicts of `epochs` or `raw` objects.
+
+Partial EEGLAB/MNE translation table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+============================================================================================================================
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: take a look at `doc/manual/io.rst` how a table can be formatted. It's being rendered incorrectly at the moment.

Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: Yeah, I did it in Pandoc Markdown format for now.

These documents are also PEP8 checked, right? So we have a line length limit of 80 still?

File: doc/manual/migrating_from_eeglab.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
+Caveats coming from EEGLAB
+==========================
+
+EEGLAB vs. MNE: Pros and Cons
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: this section is interesting to read (for me) but feels a bit verbose. I'll let @agramfort comment on the contents.

File: doc/manual/migrating_from_eeglab.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+* does not have as broad *data system format support* for EEG data formats as EEGLAB does with `fileio`. However, MNE can read many common file formats, including the EEGLAB format, and of course many MEG formats.
+* does not have the extensive inbuilt *single-trial* capabilities of EEGLAB. For example, while it is possible to plot single trials sorted by alpha power by feeding mne.viz.plot_epochs_image a function built around psd_epochs, it requires much more user input.
+* does not easily allow the rich storage of epochs-level *event information* EEGLAB has. MNE events are stored in the three-column `events` format, all integers.
+* does not have an equivalent to EEGLAB's *STUDY* structure. Grand averages of single-subject `evoked` and `average_tfr` objects can be constructed with the `mne.grand_average` function (beware of the default setting of weighting by trials), which return regular `evoked` and `average_tfr` objects. Individual subjects can simply be stored in lists or dicts of `epochs` or `raw` objects.
+
+Partial EEGLAB/MNE translation table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I think I like this part. If you can format it properly, I think this would be a great addition

File: doc/manual/migrating_from_eeglab.rst (hunk 2 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: you don't need pep8 for these. It's only for python code. However sometimes, people prefer to break the lines so that git diffs look more clean.

This will be of help: http://sphinx-doc.org/rest.html

File: doc/manual/migrating_from_eeglab.rst (hunk 4 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: It was your suggestion to have it :)

File: doc/manual/migrating_from_eeglab.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                              Software
+Concept                       EEGLAB          			        MNE
+Importing data		`EEG = pop_fileio('/your/data.dat');`		raw = mne.io.read_raw('/your/data-raw.fif')
+
+Filtering raw data	`EEG = pop_eegfiltnew(EEG, 1, 40);`			raw.filter(1, 40)
+
+Run ICA				`EEG = pop_runica(EEG);`					ica = mne.preprocessing.ica.ica().fit(raw)
+
+
+Epoching data		`EEG_epochs = pop_epoch(EEG,				`epochs = mne.Epochs(raw,
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: what about plotting epochs?

Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: You mean, interactive plotting of epochs for e.g. artefact detection?

File: doc/manual/migrating_from_eeglab.rst [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+* does not have the extensive inbuilt *single-trial* capabilities of EEGLAB. For example, while it is possible to plot single trials sorted by alpha power by feeding mne.viz.plot_epochs_image a function built around psd_epochs, it requires much more user input.
+* does not easily allow the rich storage of epochs-level *event information* EEGLAB has. MNE events are stored in the three-column `events` format, all integers.
+* does not have an equivalent to EEGLAB's *STUDY* structure. Grand averages of single-subject `evoked` and `average_tfr` objects can be constructed with the `mne.grand_average` function (beware of the default setting of weighting by trials), which return regular `evoked` and `average_tfr` objects. Individual subjects can simply be stored in lists or dicts of `epochs` or `raw` objects.
+
+Partial EEGLAB/MNE translation table
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+============================================================================================================================
+                              Software
+Concept                       EEGLAB          			        MNE
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I'd say MNE-Python

File: doc/manual/migrating_from_eeglab.rst (hunk 5 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: yeah, `epochs.plot()`. I think EEGLAB has something similar. @jaeilepp 's project was to add it to MNE-Python

File: doc/manual/migrating_from_eeglab.rst [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Importing data		`EEG = pop_fileio('/your/data.dat');`		raw = mne.io.read_raw('/your/data-raw.fif')
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: you don't mention the EEGLAB raw/epochs reader anywhere. That's the first thing needed for migrating :)

File: doc/manual/migrating_from_eeglab.rst [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+* does **not feature a GUI**. MNE always requires explicit code. This can be frustrating for beginners, but in the long run, it supports good scientific practices - better reproducible, more scalable analyses. Generally speaking, there is some trade-off between the lack of GUI in MNE and the Python base; while GUIs are more accessible to newcomers, Python more easily allows experienced users to create custom and advanced analysis pipelines and techniques.
+* has a different *status and style* of the project. EEGLAB is the biggest EEG analysis package. MNE is a comparatively young project. It follows modern coding standards (including code review and unit tests). Its development is less monolithic, more distributed and democratic than that of EEGLAB. There are many published papers using EEGLAB, and very few with MNE-Python, and many users of EEGLAB and few of MNE.
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: i would rephrase the last sentences.

Its development is less monolithic, more distributed and collaborative than that of EEGLAB. There are many published papers using EEGLAB, and less with MNE-Python. The user base of EEGLAB is so far bigger than MNE, especially among labs working with EEG data only
.

File: doc/manual/migrating_from_eeglab.rst [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+Minor differences: MNE-Python ...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* is much more developed with regards to **MVPA/decoding analyses**. It is also developed with a close eye on the leading Machine Learning toolbox *scikit-learn*.
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: scikit-learn and also make it a link

Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: yeah, and now we have a tutorial on decoding too which you can link to

File: doc/manual/migrating_from_eeglab.rst [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+* 's *ICA is not as deeply integrated* into the system. Support for clustering ICs is very rudimentary. However, MNE's ICA implementations are fully featured, and in some ways superior to EEGLAB. For example, MNE's infomax implementation can be much faster than EEGLAB's, in large part due to multicore usage. Also, MNE ICA objects allow easily applying the same ICA fit to multiple data sets (e.g., it is easy to fit ICA on high pass filtered data and apply it to unfiltered data).
+* does not feature *plugins* as strongly as EEGLAB does. No true equivalents exist for high quality plugins such as LIMO or ERPLAB. Some features which are plugins in EEGLAB are parts of base MNE, such as CORRMAP, and many plugins are at least partially mirrored by MNE core functionality (although ironically, not Alex Gramfort's graph cut plugin for single trial analysis!!!).
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: :)

Because I used a stupid GPL license at that time and I have no time to rewrite it ...

File: doc/manual/migrating_from_eeglab.rst [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+* does not have as broad *data system format support* for EEG data formats as EEGLAB does with `fileio`. However, MNE can read many common file formats, including the EEGLAB format, and of course many MEG formats.
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: what is that?

Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: What do you mean?

Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: ok got it but I would rephrase.

does not have an extensive support of the various EEG *data formats* as EEGLAB has with `fileio`. However, MNE can read many common file formats, including the EEGLAB format, and of course many MEG formats.

Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: if you want, you could have a table comparison. It might be simpler.

File: doc/manual/migrating_from_eeglab.rst [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+* The MNE resampling function is much slower than the EEGLAB equivalent. As a temporary work-around, many MNE functions allow a `decim` keyword to downsize analyses or objects. Also, remember that MNE can be more readily parallelized than MNE; via e.g. joblib, a whole experiment can be downsampled simultaneously.
+* The MNE `eog` channel type is generally conceptualized around bipolar eye channels. Channels marked as `eog` are not included in many analyses. Typical EEGLAB behavior is roughly equivalent to setting all EEG channels, including monopolar eye sensors, to channel type `eeg`.
+* MNE evoked objects support arithmetic; you can do `(epochs["critical"].average() - epochs["control"].average()).plot()`. However note that this is weighted by trial numbers! (Use `combine_evoked` to use uniform weights.) Probably the best way to plot multi-channel ERPs in MNE is `evoked.plot(spatial_colors=True)`.
+* To recreate the ERPimages sorted by response time typical of EEGLAB, reaction times can be found with the ... command, and stored in the second column of an `events` matrix. The plot can then be created with with `mne.viz.plot_epochs_image(epochs, ... order=events[:,1].argsort())`.
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: XXX

File: doc/manual/migrating_from_eeglab.rst [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+* Beware that `mne.viz.plot_epochs_image` will by default plot all channels; use the `picks` kwarg, unless you want 128 figures popping up on your screen.
+* no true equivalent to the summary-style plots from .....  exist.
+* if you are annoyed by `mne.Epochs` giving too extensive feedback during e.g. baselining, control the logging verbosity with mne.set_log_level().
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: looks like you need to rephrase things a bit ;)

File: doc/manual/migrating_from_eeglab.rst (hunk 11 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: A link to the part of the manual where supported data type formats are shown?

File: doc/manual/migrating_from_eeglab.rst [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+* The MNE `eog` channel type is generally conceptualized around bipolar eye channels. Channels marked as `eog` are not included in many analyses. Typical EEGLAB behavior is roughly equivalent to setting all EEG channels, including monopolar eye sensors, to channel type `eeg`.
+* MNE evoked objects support arithmetic; you can do `(epochs["critical"].average() - epochs["control"].average()).plot()`. However note that this is weighted by trial numbers! (Use `combine_evoked` to use uniform weights.) Probably the best way to plot multi-channel ERPs in MNE is `evoked.plot(spatial_colors=True)`.
+* To recreate the ERPimages sorted by response time typical of EEGLAB, reaction times can be found with the ... command, and stored in the second column of an `events` matrix. The plot can then be created with with `mne.viz.plot_epochs_image(epochs, ... order=events[:,1].argsort())`.
+* Beware that `mne.viz.plot_epochs_image` will by default plot all channels; use the `picks` kwarg, unless you want 128 figures popping up on your screen.
+* no true equivalent to the summary-style plots from .....  exist.
```
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: can you share an example?

Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: I think I'll make an issue about it actually.

File: doc/manual/migrating_from_eeglab.rst (hunk 6 above)
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: progressively I would like to avoid differentiating MNE and MNE-Python. So please keep MNE

File: doc/manual/migrating_from_eeglab.rst [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+* The MNE event system stores onset, duration, and type (as integers). Beware of the find_events function; with its default settings, it will not find many events in typical EEG formats. Try using `shortest_event=0, min_duration=0, consecutive=True` if that happens to you.
+* The MNE resampling function is much slower than the EEGLAB equivalent. As a temporary work-around, many MNE functions allow a `decim` keyword to downsize analyses or objects. Also, remember that MNE can be more readily parallelized than MNE; via e.g. joblib, a whole experiment can be downsampled simultaneously.
+* The MNE `eog` channel type is generally conceptualized around bipolar eye channels. Channels marked as `eog` are not included in many analyses. Typical EEGLAB behavior is roughly equivalent to setting all EEG channels, including monopolar eye sensors, to channel type `eeg`.
+* MNE evoked objects support arithmetic; you can do `(epochs["critical"].average() - epochs["control"].average()).plot()`. However note that this is weighted by trial numbers! (Use `combine_evoked` to use uniform weights.) Probably the best way to plot multi-channel ERPs in MNE is `evoked.plot(spatial_colors=True)`.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I would just say:

`MNE evoked objects support arithmetic (with a catch)`

and link "with a catch" to `pitfalls.rst`

File: doc/manual/migrating_from_eeglab.rst (hunk 14 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: #2747

File: doc/manual/migrating_from_eeglab.rst (hunk 7 above)
Comment (teonlamont) [inline doc/manual/migrating_from_eeglab.rst]: also `mne.io.read_raw` doesn't exist (yet/maybe never)

File: doc/manual/migrating_from_eeglab.rst [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+============================================================================================================================
+                              Software
+Concept                       EEGLAB          			        MNE
+Importing data		`EEG = pop_fileio('/your/data.dat');`		raw = mne.io.read_raw('/your/data-raw.fif')
+
+Filtering raw data	`EEG = pop_eegfiltnew(EEG, 1, 40);`			raw.filter(1, 40)
+
+Run ICA				`EEG = pop_runica(EEG);`					ica = mne.preprocessing.ica.ica().fit(raw)
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: This is not how you'd typically do the ica ... maybe it's better to do an import and split it into two lines.

File: doc/manual/migrating_from_eeglab.rst [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+* does not feature *plugins* as strongly as EEGLAB does. No true equivalents exist for high quality plugins such as LIMO or ERPLAB. Some features which are plugins in EEGLAB are parts of base MNE, such as CORRMAP, and many plugins are at least partially mirrored by MNE core functionality (although ironically, not Alex Gramfort's graph cut plugin for single trial analysis!!!).
+* does not have as broad *data system format support* for EEG data formats as EEGLAB does with `fileio`. However, MNE can read many common file formats, including the EEGLAB format, and of course many MEG formats.
+* does not have the extensive inbuilt *single-trial* capabilities of EEGLAB. For example, while it is possible to plot single trials sorted by alpha power by feeding mne.viz.plot_epochs_image a function built around psd_epochs, it requires much more user input.
+* does not easily allow the rich storage of epochs-level *event information* EEGLAB has. MNE events are stored in the three-column `events` format, all integers.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I'd be more specific:

`In EEGLAB, events are stored in a Matlab structure with type and latency as compulsory fields and many other optional fields. MNE events on the other hand store the events in a much simpler three column numpy array`

File: doc/manual/migrating_from_eeglab.rst (hunk 11 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: yeah, maybe. Put it in and let's see how it reads :)

File: doc/manual/migrating_from_eeglab.rst [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+============================================================================================================================
+
+
+Tips on specific features for people coming from EEGLAB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* The MNE event system stores onset, duration, and type (as integers). Beware of the find_events function; with its default settings, it will not find many events in typical EEG formats. Try using `shortest_event=0, min_duration=0, consecutive=True` if that happens to you.
+* The MNE resampling function is much slower than the EEGLAB equivalent. As a temporary work-around, many MNE functions allow a `decim` keyword to downsize analyses or objects. Also, remember that MNE can be more readily parallelized than MNE; via e.g. joblib, a whole experiment can be downsampled simultaneously.
+* The MNE `eog` channel type is generally conceptualized around bipolar eye channels. Channels marked as `eog` are not included in many analyses. Typical EEGLAB behavior is roughly equivalent to setting all EEG channels, including monopolar eye sensors, to channel type `eeg`.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I would be more explicit. Can you put in two lines of code here? Something like:

```py
In contrast to EEGLAB, you are required to specify the EOG channels in MNE as it is critical to many analysis steps including preprocessing, projection and forward calculation. This can be done when reading in the file:

raw = read_raw_eeglab('filename.set', eog=['EOG1, 'EOG2'])
```

File: doc/manual/migrating_from_eeglab.rst [hunk 19]
```diff
@@ -0,0 +1,68 @@
+Caveats coming from EEGLAB
+==========================
+
+EEGLAB vs. MNE: Pros and Cons
+-----------------------------
+
+Major differences: MNE-Python ...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* is **Python based**. Python is a fully featured, general programming language. It is easy to learn, easy to read (and thus share, criticize and reproduce code with). It is also free - unlike MATLAB, which you need for EEGLAB, and the MATLAB toolboxes. For example, with MNE, you will never run out of/lack parallel processing toolboxes, and thus MNE more readily uses multiple cores (usually with the `n_jobs` keyword).
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: I would link "easy to learn" to this page: https://learnxinyminutes.com/docs/python/

File: doc/manual/migrating_from_eeglab.rst [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+* was born as a package for **MEG analysis**. While *MNE has gained strong EEG support*, it in many instances takes a more extensive approach. For example, MEG data is usually multimodal (including magnetometers, gradiometers and EEG), and thus MNE requires you to constantly choose what channel type you want to operate with. MNE also has much more integrated support for source-level/source localisation than EEGLAB.
+* is **Object oriented**. Typical user-exposed functions don't operate on raw data matrices, but on `Raw`, `Epochs`, `Evoked`, ... objects. Many object methods operate in-place. For example, when in MNE, you do EEG = pop_eegfiltnew(EEG, ...), in MNE, you do raw.filter(...). In EEGLAB, you do pop_plot_topo(EEG, ...). In MNE, you do evoked.plot_topomap(...). In EEGLAB, you do pop_saveset(EEG, ...). In MNE, you do `epochs.save(...)`. `raw.plot()` is very different from `evoked.plot()`. In-place modification can usually be turned off with the `copy=True` setting.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: just for the record, you can do object oriented programming in Matlab: http://in.mathworks.com/help/matlab/object-oriented-programming.html. I'd be careful with that statement.

File: doc/manual/migrating_from_eeglab.rst [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+* does **not feature a GUI**. MNE always requires explicit code. This can be frustrating for beginners, but in the long run, it supports good scientific practices - better reproducible, more scalable analyses. Generally speaking, there is some trade-off between the lack of GUI in MNE and the Python base; while GUIs are more accessible to newcomers, Python more easily allows experienced users to create custom and advanced analysis pipelines and techniques.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: maybe you can point out that even though we don't support GUI, we do allow interactive exploration through our plotting routines, c.f., epochs.plot() and interactive projs etc.

File: doc/manual/migrating_from_eeglab.rst [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+
+Saving data         `EEG = pop_saveset(EEG,'filename',set);`	`[raw/epochs/evoked].save('/your/data-[raw/epo/evo].fif')
+
+============================================================================================================================
+
+
+Tips on specific features for people coming from EEGLAB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* The MNE event system stores onset, duration, and type (as integers). Beware of the find_events function; with its default settings, it will not find many events in typical EEG formats. Try using `shortest_event=0, min_duration=0, consecutive=True` if that happens to you.
+* The MNE resampling function is much slower than the EEGLAB equivalent. As a temporary work-around, many MNE functions allow a `decim` keyword to downsize analyses or objects. Also, remember that MNE can be more readily parallelized than MNE; via e.g. joblib, a whole experiment can be downsampled simultaneously.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: Maybe put one line of code to demonstrate how easy it is to parallelize?

```py
>>> raw.resample(sfreq * 2, n_jobs = -1)
```

File: doc/manual/migrating_from_eeglab.rst [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+* The MNE `eog` channel type is generally conceptualized around bipolar eye channels. Channels marked as `eog` are not included in many analyses. Typical EEGLAB behavior is roughly equivalent to setting all EEG channels, including monopolar eye sensors, to channel type `eeg`.
+* MNE evoked objects support arithmetic; you can do `(epochs["critical"].average() - epochs["control"].average()).plot()`. However note that this is weighted by trial numbers! (Use `combine_evoked` to use uniform weights.) Probably the best way to plot multi-channel ERPs in MNE is `evoked.plot(spatial_colors=True)`.
+* To recreate the ERPimages sorted by response time typical of EEGLAB, reaction times can be found with the ... command, and stored in the second column of an `events` matrix. The plot can then be created with with `mne.viz.plot_epochs_image(epochs, ... order=events[:,1].argsort())`.
+* Beware that `mne.viz.plot_epochs_image` will by default plot all channels; use the `picks` kwarg, unless you want 128 figures popping up on your screen.
```
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: this needs to be fixed if this is a problem. How do they handle it in EEGLAB?

**Comment by jasmainak** [comment]
@jona-sassenhagen let us know when you're ready for another round of review :)

File: doc/manual/migrating_from_eeglab.rst (hunk 4 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: :)

File: doc/manual/migrating_from_eeglab.rst (hunk 22 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: Also, something that I use quite often is memory caching from joblibs. I don't think there's a simple equivalent in Matlab for that.

File: doc/manual/migrating_from_eeglab.rst (hunk 8 above)
Comment (dengemann) [inline doc/manual/migrating_from_eeglab.rst]: Please be careful, we don't want to get into politics here. I would basically drop the qualifyimg sentences that say X is better or more developped, such sentences may be taken literally by many and can be outdated very quickly. It is not good for the peace among neighbours.

File: doc/manual/migrating_from_eeglab.rst (hunk 17 above)
Comment (dengemann) [inline doc/manual/migrating_from_eeglab.rst]: Yes please only list specific differences and lets avoid general comparisons.

File: doc/manual/migrating_from_eeglab.rst (hunk 16 above)
Comment (dengemann) [inline doc/manual/migrating_from_eeglab.rst]: Maybe not a good comparison, the ica class is much more low level. Then also it would be uppercase letters.

File: doc/manual/migrating_from_eeglab.rst (hunk 17 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: > lets avoid general comparisons

+1

**Comment by dengemann** [comment]
Thanks Jona, a nice initiative. Let's make sure to avoid general evaluations and instead point out specific differences.

File: doc/manual/migrating_from_eeglab.rst (hunk 8 above)
Comment (agramfort) [inline doc/manual/migrating_from_eeglab.rst]: I agree maybe remove the last to sentence and just say

Its development collaborative and distributed across many labs world wide.

File: doc/manual/migrating_from_eeglab.rst (hunk 16 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: pop_runica is as low-level as it gets though, it only fits ICA.

File: doc/manual/migrating_from_eeglab.rst (hunk 20 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: The point is that *EEGLAB* is not object oriented, but *MNE* is. You can, after all, also work very functional or imperative in Python. But one thing you'll notice coming from EEGLAB to MNE is MNE being object oriented.

File: doc/manual/migrating_from_eeglab.rst (hunk 18 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: I think for properly `indoctrinated` EEGLAB users, it won't be necessary to introduce the concept of EOG channels for the reason that they're not good practice in the first place.

File: doc/manual/migrating_from_eeglab.rst (hunk 23 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: The EEGLAB functions for stuff like this all require you to pre-specify the single channel (or component) you want to plot.

**Comment by jona-sassenhagen** [comment]
I've started a feature comparison table, with plusses and minuses for stuff like "decoding capabilities", "time-frequency capabilities", but I'm getting the impression that's not what we're going for here?

File: doc/manual/migrating_from_eeglab.rst (hunk 20 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: yeah, but that's not correct. From the EEGLAB documentation: http://sccn.ucsd.edu/wiki/The_first_decade_of_EEGLAB

> At the same time, a developmental version of EEGLAB v10 was released including new features such as an optional object-oriented EEG structure.

File: doc/manual/migrating_from_eeglab.rst (hunk 23 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: So, you mean picks should be made compulsory argument instead of optional? Why not. Please raise an issue and remove this line then.

File: doc/manual/migrating_from_eeglab.rst (hunk 20 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: The primary EEGLAB code and the stuff they teach at seminars is not object oriented. See e.g. http://sccn.ucsd.edu/mediawiki/images/e/e6/EEGLAB2015_Sheffield_AD_STUDYscripting.pdf

It's all `EEG = pop_somefunction(EEG, params);` etc. The 'eegobj' thing from v10 never got any traction from what I can tell (outside some of the less-used, more experimental plugins, which are object oriented).

File: doc/manual/migrating_from_eeglab.rst (hunk 23 above)
Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: Oh god, I have so many open issues :)

File: doc/manual/migrating_from_eeglab.rst (hunk 18 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: Umm ...  if you have only MEG, would you treat the EOG channels as a separate channel type or as EEG? How would you reject eyeblinks in that case?

Comment (jona-sassenhagen) [inline doc/manual/migrating_from_eeglab.rst]: Coming from EEGLAB, via ICA.
I can't say anything about MEG though.

File: doc/manual/migrating_from_eeglab.rst (hunk 20 above)
Comment (jasmainak) [inline doc/manual/migrating_from_eeglab.rst]: hmm ... yeah. But let's be diplomatic and try to avoid statements which can be contradicted :)

**Comment by jasmainak** [comment]
@jona-sassenhagen I've made a PR here: https://github.com/mne-tools/mne-python/pull/2961 since this had been untouched for a while. Comments welcome
